PowerEditor/src/NppIO.cpp

```diff
@@ -1756,12 +1756,27 @@ bool Notepad_plus::fileRename(BufferID id)
 	{
 		CustomFileDialog fDlg(_pPublicInterface->getHSelf());
 
-		fDlg.setExtFilter(TEXT("All types"), TEXT(".*"));
+		for (auto item : ScintillaEditView::langNames)
+		{
+			if (bufLang == item.LangID)
+			{
+				fDlg.setExtFilter(item.longName, item.lexerName);
+			}
+			else
+			{
+				fDlg.setExtFilter(TEXT("All types"), TEXT(".*"));
+			}
+		}

```

Comment:
Should the above construct look more like below? I haven't tested it, it's just a feeling. Your current algorithm sets a filter with every loop iteration, which doesn't look like desired, because in fact, only the last filter item is set then, isn't it?

```
fDlg.setExtFilter(TEXT("All types"), TEXT(".*"));
for (auto item : ScintillaEditView::langNames)
{
	if (bufLang == item.LangID)
	{
		fDlg.setExtFilter(item.longName, item.lexerName);
		break;
	}
}
```